### PR TITLE
Changed configuration assessment name to SCA

### PR DESCRIPTION
**Pull request by TJOSERAFAEL**
### Changed module name

The configuration assessment module is now renamed to sca (security configuration assessment). This is due to the fact that the name was too long.

### CODE REVIEW

File: src/config/config.c [hunk 1]
```diff
@@ -51,7 +51,7 @@ static int read_main_elements(const OS_XML *xml, int modules,
     const char *oslogging = "logging";                  /* Logging Config */
     const char *oscluster = "cluster";                  /* Cluster Config */
     const char *ossocket = "socket";                    /* Socket Config */
-    const char *osconfiguration_assessment = "configuration-assessment"; /* Wodle - Configuration Assessment  */
+    const char *osconfiguration_assessment = "sca"; /* Wodle - Security Configuration Assessment  */
```
Comment (chemamartinez) [inline src/config/config.c]: Remove the wodle tag

File: src/init/inst-functions.sh [hunk 2]
```diff
@@ -234,10 +234,10 @@ InstallConfigurationAssessmentFiles()
         echo "Installing configuration assessment policies..."
         CONFIGURATION_ASSESSMENT_FILES=$(cat .$CONFIGURATION_ASSESSMENT_FILES_PATH)
         for file in $CONFIGURATION_ASSESSMENT_FILES; do
-            if [ -f "../etc/configuration-assessment/$file" ]; then
-                ${INSTALL} -v -m 0640 -o root -g ${OSSEC_GROUP} ../etc/configuration-assessment/$file ${PREFIX}/ruleset/configuration-assessment
+            if [ -f "../etc/sca/$file" ]; then
+                ${INSTALL} -v -m 0640 -o root -g ${OSSEC_GROUP} ../etc/sca/$file ${PREFIX}/ruleset/sca
             else
-                echo "ERROR: Configuration assessment policy not found: ./etc/configuration-assessment/$file"
+                echo "ERROR: Configuration assessment policy not found: ./etc/sca/$file"
```
Comment (chemamartinez) [inline src/init/inst-functions.sh]: change the message to `SCA policy not found`

File: src/rootcheck/rootcheck.c [hunk 3]
```diff
@@ -170,6 +170,16 @@ int rootcheck_init(int test_config)
         mterror_exit(ARGV0, CONFIG_ERROR, cfg);
     }
 
+    if(rootcheck.checks.rc_unixaudit) {
+        mwarn("The check_unixaudit option is deprecated in favor of the Configuration Assessment module.");
```
Comment (chemamartinez) [inline src/rootcheck/rootcheck.c]: `the Configuration Assessment module` -> `the SCA module`

File: src/rootcheck/rootcheck.c [hunk 4]
```diff
@@ -170,6 +170,16 @@ int rootcheck_init(int test_config)
...
+    }
+
+#ifdef WIN32
+    if(rootcheck.checks.rc_winaudit) {
+        mwarn("The check_winaudit option is deprecated in favor of the Configuration Assessment module.");
```
Comment (chemamartinez) [inline src/rootcheck/rootcheck.c]: the SCA module

File: src/wazuh_db/wdb_sca.c [hunk 5]
```diff
@@ -170,20 +170,20 @@ int wdb_configuration_assessment_save(wdb_t * wdb, int id,int scan_id,char * tit
 }
 
 /* Update global configuration assessment entry. Returns number of affected rows or -1 on error.  */
-int wdb_configuration_assessment_global_update(wdb_t * wdb, int scan_id, char *name,char *description,char *references,int pass,int failed,int score) {
+int wdb_sca_global_update(wdb_t * wdb, int scan_id, char *name,char *description,char *references,int pass,int failed,int score) {
     if (!wdb->transaction && wdb_begin2(wdb) < 0){
         mdebug1("at wdb_rootcheck_save(): cannot begin transaction");
```
Comment (chemamartinez) [inline src/wazuh_db/wdb_sca.c]: `wdb_rootcheck_save`-> `wdb_sca_save`. It is repeated several times

File: src/rootcheck/rootcheck.c [hunk 6]
```diff
@@ -170,6 +170,16 @@ int rootcheck_init(int test_config)
         mterror_exit(ARGV0, CONFIG_ERROR, cfg);
     }
 
+    if(rootcheck.checks.rc_unixaudit) {
```
Comment (chemamartinez) [inline src/rootcheck/rootcheck.c]: This cannot works when building the Windows agent:
```
rootcheck/rootcheck.c: In function ‘rootcheck_init’:
rootcheck/rootcheck.c:173:25: error: ‘struct _checks’ has no member named ‘rc_unixaudit’; did you mean ‘rc_winaudit’?
     if(rootcheck.checks.rc_unixaudit) {
                         ^~~~~~~~~~~~
                         rc_winaudit
```

File: src/wazuh_modules/wm_sca.c [hunk 7]
```diff
@@ -844,13 +845,13 @@ static int wm_configuration_assessment_do_scan(OSList *p_list,cJSON *profile_che
                     char *f_value = NULL;
                     char *dir = NULL;
 
-                    file = wm_configuration_assessment_get_pattern(value);
+                    file = wm_sca_get_pattern(value);
                     if (!file) {
                         merror(WM_CONFIGURATION_ASSESSMENT_MONITORING_INVALID_RKCL_VAR, value);
```
Comment (chemamartinez) [inline src/wazuh_modules/wm_sca.c]: `WM_CONFIGURATION_ASSESSMENT_MONITORING_INVALID_RKCL_VAR` -> `WM_SCA_INVALID_VAR` for example

**Comment by chemamartinez** [comment]
Apart from previous comments, it remains the following points:

- The changelog entry.
- The module name in the elastic template.
- The templates file name: https://github.com/wazuh/wazuh/tree/3.9-sca/etc/templates/config/generic to `sca_generic.template`, `sca.files`, etc.
- Changes to the previous templates names in the `inst-functions.sh` functions.

File: src/win32/ossec.conf [hunk 8]
```diff
@@ -57,7 +57,7 @@
   </rootcheck>
 
   <!-- Configuration Assessment -->
-  <configuration-assessment>
+  <sca>
```
Comment (vikman90) [inline src/win32/ossec.conf]: Save this file as LF.

File: .travis.yml [hunk 9]
```diff
@@ -11,7 +11,7 @@ env:
 - USER_ENABLE_ROOTCHECK="y"
 - USER_ENABLE_OPENSCAP="y"
 - USER_ENABLE_SYSCOLLECTOR="y"
-- USER_ENABLE_CONFIGURATION_ASSESSMENT="y"
+- USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT="y"
```
Comment (vikman90) [inline .travis.yml]: Let's use `USE_ENABLE_SCA`.

File: contrib/specs/agent/preloaded-vars.conf [hunk 10]
```diff
@@ -67,10 +67,10 @@ USER_ENABLE_ROOTCHECK="y"
 # disable it.
 USER_ENABLE_SYSCOLLECTOR="y"
 
-# If USER_ENABLE_CONFIGURATION_ASSESSMENT is set to "y",
+# If USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT is set to "y",
 # Configuration assessment will be enabled. Set to "n" to
 # disable it.
-USER_ENABLE_CONFIGURATION_ASSESSMENT="y"
+USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT="y"
```
Comment (vikman90) [inline contrib/specs/agent/preloaded-vars.conf]: `USER_ENABLE_SCA`

File: contrib/specs/local/preloaded-vars.conf [hunk 11]
```diff
@@ -67,10 +67,10 @@ USER_ENABLE_ROOTCHECK="y"
 # disable it.
 USER_ENABLE_SYSCOLLECTOR="y"
 
-# If USER_ENABLE_CONFIGURATION_ASSESSMENT is set to "y",
-# Configuration assessment will be enabled. Set to "n" to
+# If USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT is set to "y",
+# Security configuration assessment will be enabled. Set to "n" to
 # disable it.
-USER_ENABLE_CONFIGURATION_ASSESSMENT="y"
+USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT="y"
```
Comment (vikman90) [inline contrib/specs/local/preloaded-vars.conf]: `USER_ENABLE_SCA`

File: contrib/specs/server/preloaded-vars.conf [hunk 12]
```diff
@@ -67,10 +67,10 @@ USER_ENABLE_ROOTCHECK="y"
 # disable it.
 USER_ENABLE_SYSCOLLECTOR="y"
 
-# If USER_ENABLE_CONFIGURATION_ASSESSMENT is set to "y",
-# Configuration assessment will be enabled. Set to "n" to
+# If USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT is set to "y",
+# Security configuration assessment will be enabled. Set to "n" to
 # disable it.
-USER_ENABLE_CONFIGURATION_ASSESSMENT="y"
+USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT="y"
```
Comment (vikman90) [inline contrib/specs/server/preloaded-vars.conf]: `USER_ENABLE_SCA`

File: etc/preloaded-vars.conf [hunk 13]
```diff
@@ -75,10 +75,10 @@
 # disable it.
 #USER_ENABLE_SYSCOLLECTOR="y"
 
-# If USER_ENABLE_CONFIGURATION_ASSESSMENT is set to "y",
+# If USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT is set to "y",
 # the configuration assessment module will be enabled.
 # Set to "n" to disable it.
-USER_ENABLE_CONFIGURATION_ASSESSMENT="y"
+USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT="y"
```
Comment (vikman90) [inline etc/preloaded-vars.conf]: `USER_ENABLE_SCA`

File: install.sh [hunk 14]
```diff
@@ -265,13 +265,13 @@ UseSyscollector()
      fi
 }
 
-UseConfigurationAssessment()
+UseSecurityConfigurationAssessment()
 {
     # Configuration assessment config predefined (is overwritten by the preload-vars file)
-    if [ "X${USER_ENABLE_CONFIGURATION_ASSESSMENT}" = "Xn" ]; then
-        CONFIGURATION_ASSESSMENT="no"
+    if [ "X${USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT}" = "Xn" ]; then
```
Comment (vikman90) [inline install.sh]: `USER_ENABLE_SCA`

File: src/analysisd/analysisd.c [hunk 15]
```diff
@@ -59,7 +59,7 @@ int DecodeHostinfo(Eventinfo *lf);
 int DecodeSyscollector(Eventinfo *lf,int *socket);
 int DecodeCiscat(Eventinfo *lf);
 int DecodeWinevt(Eventinfo *lf);
-int DecodeConfigurationAssessment(Eventinfo *lf,int *socket);
+int DecodeSecurityConfigurationAssessment(Eventinfo *lf,int *socket);
```
Comment (vikman90) [inline src/analysisd/analysisd.c]:
```c
int DecodeSCA();
```

File: src/analysisd/analysisd.c [hunk 16]
```diff
@@ -1617,25 +1617,25 @@ void * ad_input_main(void * args) {
 
                 /* Increment number of events received */
                 hourly_events++;
-            } else if(msg[0] == CONFIGURATION_ASSESSMENT_MQ){
+            } else if(msg[0] == SECURITY_CONFIGURATION_ASSESSMENT_MQ){
```
Comment (vikman90) [inline src/analysisd/analysisd.c]: `SCA_MQ`

File: src/analysisd/decoders/decode-xml.c [hunk 17]
```diff
@@ -779,7 +779,7 @@ int SetDecodeXML()
     addDecoder2list(SYSCOLLECTOR_MOD);
     addDecoder2list(CISCAT_MOD);
     addDecoder2list(WINEVT_MOD);
-    addDecoder2list(CONFIGURATION_ASSESSMENT_MOD);
+    addDecoder2list(SECURITY_CONFIGURATION_ASSESSMENT_MOD);
```
Comment (vikman90) [inline src/analysisd/decoders/decode-xml.c]: `SCA_MOD`

File: src/config/wmodules-config.c [hunk 18]
```diff
@@ -146,7 +146,7 @@ int Read_WModule(const OS_XML *xml, xml_node *node, void *d1, void *d2)
     return 0;
 }
 
-int Read_ConfigurationAssessment(const OS_XML *xml, xml_node *node, void *d1)
+int Read_SecurityConfigurationAssessment(const OS_XML *xml, xml_node *node, void *d1)
```
Comment (vikman90) [inline src/config/wmodules-config.c]:
```c
Read_SCA();
```

File: src/config/wmodules-sca.c [hunk 19]
```diff
@@ -26,33 +26,34 @@ static short eval_bool(const char *str)
...
 
 // Reading function
-int wm_configuration_assessment_read(const OS_XML *xml,xml_node **nodes, wmodule *module)
+int wm_sca_read(const OS_XML *xml,xml_node **nodes, wmodule *module)
 {
     unsigned int i;
     unsigned int profiles = 0;
     int month_interval = 0;
-    wm_configuration_assessment_t *configuration_assessment;
+    wm_sca_t *security_configuration_assessment;
```
Comment (vikman90) [inline src/config/wmodules-sca.c]:
```c
wm_sca_t * sca;
```

**Pull request merged**